app/core/state_machines.py

```
@@ -0,0 +1,391 @@
+
+        Args:
+            state: The state to check.
+
+        Returns:
+            True if the state is terminal, False otherwise.
+
+        Raises:
+            ValueError: If the state is not a known AttackState.
+        """
+        if state not in cls.TRANSITIONS:
+            raise ValueError(f"Unknown attack state: {state}")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)